### DISCUSSION TITLE
Is it possible to stream the audio while it is being generated?

### DISCUSSION THREAD

**Discussion by AurisDo** [rhasspy/piper#298] [Q&A]
Can anyone please tell me that we can read the audio generated by Piper in stream? I wonder why we have to wait for the audio to be generated completely and then saving to a file and finally playing the file, while we can just stream the audio directly when it is being generated?

**Comment by synesthesiam** [comment]
https://github.com/rhasspy/piper#streaming-audio

**Reply by AurisDo** [reply]
Thanks for the answer!!!!!!

**Reply by AurisDo** [reply]
Hello Sir, when I run the command line it said: 'aplay' is not recognized as an internal or external command, how can I fix this? Thanks!

**Reply by sweetbbak** [reply]
you either need aplay, pacat or ffplay from FFMPEG, then use S16_LE single channel audio like this:
```
pacat --rate=22050 --format=s16le --channels=1 --raw
aplay -r 22050 -c 1 -f S16_LE -t raw
ffplay -f s16le -ar 22.05k -ac 1 -nodisp -autoexit
```
all of these read audio from stdin, so just run piper and pipe it into one of these commands. You can install ffmpeg anywhere and aplay is from alsa-utils and pacat is from pulseaudio utils. All distros will have this, you could probably use WSL if you needed to as well.

**Comment by rmcpantoja** [comment]
You can also use #255.

**Reply by AurisDo** [reply]
Thanks for your recommendation!
